test/e2e/serial/tests/configuration.go

```diff
@@ -1172,3 +1460,45 @@ func isCustomPolicySupportEnabled(nro *nropv1.NUMAResourcesOperator) bool {
 	}
 	return inthelper.IsCustomPolicyEnabled(nro)
 }
+
+func createTAEDeployment(fxt *e2efixture.Fixture, ctx context.Context, name, schedulerName, cpus string) *appsv1.Deployment {
+	var err error
+	var replicas int32 = 1
+
+	podLabels := map[string]string{
+		"test": "test-dp",
+	}
+	nodeSelector := map[string]string{}
+	deployment := objects.NewTestDeployment(replicas, podLabels, nodeSelector, fxt.Namespace.Name, name, images.GetPauseImage(), []string{images.PauseCommand}, []string{})
+	deployment.Spec.Template.Spec.SchedulerName = schedulerName
+	deployment.Spec.Template.Spec.Containers[0].Resources.Limits = corev1.ResourceList{
+		corev1.ResourceCPU:    resource.MustParse(cpus),
+		corev1.ResourceMemory: resource.MustParse("256Mi"),
+	}
+
+	By(fmt.Sprintf("creating a topology affinity error deployment %q", name))
+	err = fxt.Client.Create(ctx, deployment)
+	Expect(err).ToNot(HaveOccurred())
+
+	return deployment
+}
+
+// updatePerformanceProfileFieldUnstructured updates a specific field in a PerformanceProfile using the dynamic client.
+// It takes the field path as variadic string arguments and updates the field with the provided value.
+func updatePerformanceProfileFieldUnstructured(ctx context.Context, dynamicClient dynamic.Interface, name string, value interface{}, fieldPath ...string) error {
+	performanceProfileGVR := schema.GroupVersionResource{
+		Group:    "performance.openshift.io",
+		Version:  "v2",
+		Resource: "performanceprofiles",
+	}
+	profile, err := dynamicClient.Resource(performanceProfileGVR).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	err = unstructured.SetNestedField(profile.Object, value, fieldPath...)
+	if err != nil {
+		return err
```

Comment:
let's rephrase the error and append the original one within a new more clear one pointing to the set operation that failed.